client/main.lua

```diff
@@ -264,23 +265,58 @@ end)
 --Check distances
 local function checkTakeDist(pos, loc, garage, ped, type, indexgarage)
     local takeDist = #(pos - vector3(loc.x, loc.y, loc.z))
-    if takeDist <= 15 then
-        inGarageRange = true
-        DrawMarker(2, loc.x, loc.y, loc.z, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.3, 0.2, 0.15, 200, 0, 0, 222, false, false, false, true, false, false, false)
-        if takeDist <= 1.5 then
-            if not IsPedInAnyVehicle(ped) then
-                if type == "house" then
-                    DrawText3Ds(loc.x, loc.y, loc.z + 0.5, Lang:t("info.car_e"))
-                else
-                    DrawText3Ds(loc.x, loc.y, loc.z + 0.5, Lang:t("info."..garage.vehicle.."_e"))
+    
+    if takeDist >= 85 then -- Free up entity memory
+        if NpcData[indexgarage] then
+            if NpcData[indexgarage].ped then
+                SetEntityAsNoLongerNeeded(NpcData[indexgarage].ped)
```

Comment:
This doesn't remove the ped

Reply:
Yes I saw that they were just leaving the area and if you have a no NPC script enabled they will despawn. I guess with normal NPCs on they won't despawn aswell.

I just saw some script using `DeletePed()` but I have to check that out myself first.

Reply:
> Yes I saw that they were just leaving the area and if you have a no NPC script enabled they will despawn. I guess with normal NPCs on they won't despawn aswell.
> 
> I just saw some script using `DeletePed()` but I have to check that out myself first.

will you be updating this?

Reply:
Yep sorry, I was working hard on my server for the release. I will take some time today or tomorrow to update and test it.

Reply:
This might take a little bit more time as I have to first get into the new PolyZone logic. Will take a look at it this afternoon.